### PR TITLE
fix

### PR DESCRIPTION
**Pull request by Soyn**
I think there is not necessary to assign the nullptr. We will assign the `rhs.value` to the left,right and count.

### CODE REVIEW

**Comment by pezy** [comment]
Agree. 

@Mooophy

**Comment by Mooophy** [comment]
Seems @Soyn has added more commits, can you make a double check. If no problem, you can merge it. @pezy

File: ch13/ex13_30.h [hunk 1]
```diff
@@ -18,7 +18,14 @@
...
-    friend void swap(HasPtr&, HasPtr&);
+    friend inline void swap(HasPtr& lhs, HasPtr& rhs)
+    {
+    using std::swap;
+    swap(lhs.ps, rhs.ps);
+    swap(lhs.i, rhs.i);
+    std::cout << "call swap(HasPtr& lhs, HasPtr& rhs)" << std::endl;
+    }
+    
     HasPtr(const std::string &s = std::string()) : ps(new std::string(s)), i(0) { }
```
Comment (pezy) [inline ch13/ex13_30.h]: @Soyn  Please keep the **indentation**, and the `inline` there is **redundant**.

**Comment by Soyn** [comment]
I find the swap() should be inline. In page457~458, the book tell us to use the swap() to optimization the code, so i commit it.

**Comment by pezy** [comment]
> I find the swap() should be inline.

why ?

**Comment by Soyn** [comment]
the swap()'s existence is to optimization the code, so i think you lost the inline.You can check this in Page458(Chapter13.3)

**Comment by Soyn** [comment]
If I make a mistake,I close it, then pull a new request.

**Comment by Mooophy** [comment]
Please contact me via WeChat.

**Comment by Mooophy** [comment]
WeChat id = moooophy
@Soyn

**Pull request closed without merging**